### PR TITLE
update download_dataset_files api

### PR DESCRIPTION
**Pull request by ihpolash**
update download_dataset_files api

### CODE REVIEW

File: python/bagel/api/fastapi.py [hunk 1]
```diff
@@ -653,6 +653,37 @@ def download_dataset(
...
+                file_content, file_name, file_type = self.download_dataset(
+                    dataset_id=dataset_id,
+                    file_path=file_path
+                )
+
+                file_path = os.path.join(target_dir, file_name)
+                with open(file_path, "wb") as file:
+                    file.write(file_content)
+
+            elif file_info['type'] == "dir":
```
Comment (Rijul1204) [inline python/bagel/api/fastapi.py:682]: make it constant variable

Comment (ihpolash) [reply python/bagel/api/fastapi.py]: made the changes

**Comment by ihpolash** [comment]
@Rijul1204 Updated with PR title and commit message accordingly. Could you please make another review?

File: python/bagel/api/__init__.py [hunk 2]
```diff
@@ -457,4 +457,14 @@ def download_dataset(
...
         pass
+    
+    @abstractmethod
+    def download_dataset_files(
+            self,
+            dataset_id: str, 
+            target_dir: str,
+            file_path: Optional[str] = "",
+            api_key: Optional[str] = None
+            ) -> str:
```
Comment (Rijul1204) [inline python/bagel/api/__init__.py:468]: Same here

File: python/bagel/api/fastapi.py [hunk 3]
```diff
@@ -653,6 +653,44 @@ def download_dataset(
...
         return file_content, file_name, file_type
+    
+    @override
+    def download_dataset_files(
+            self, 
+            dataset_id: str, 
+            target_dir: str,
+            file_path: Optional[str] = "",
+            api_key: Optional[str] = None
+            ) -> str:
```
[the remark this answers is not in the document]
Comment (ihpolash) [reply python/bagel/api/fastapi.py]: Fixed it!

File: python/bagel/api/__init__.py (hunk 2 above)
Comment (ihpolash) [reply python/bagel/api/__init__.py]: Fixed it!

**Review by Rijul1204 — APPROVED** [review]

**Pull request merged**